cinema/views.py

```diff
@@ -1,45 +1,107 @@
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+class GenreDetail(views.APIView):
+    def get_object(self, pk: int) -> Genre:
+        return get_object_or_404(Genre, pk=pk)
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
-
-    if request.method == "GET":
-        serializer = MovieSerializer(movie)
+    def get(self, request: Request, pk: int) -> Response:
+        serializer = GenreSerializer(self.get_object(pk))
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "PUT":
-        serializer = MovieSerializer(movie, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+    def put(self, request: Request, pk: int) -> Response:
+        serializer = GenreSerializer(self.get_object(pk), data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    def patch(self, request: Request, pk: int) -> Response:
+        genre = self.get_object(pk=pk)
+        serializer = GenreSerializer(genre, data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "DELETE":
-        movie.delete()
+    def delete(self, request: Request, pk: int) -> Response:
+        genre = self.get_object(pk=pk)
+        genre.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class ActorList(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    generics.GenericAPIView,
+):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get(self, request, *args, **kwargs) -> Response:
+        return self.list(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs) -> Response:
+        return self.create(request, *args, **kwargs)
+
+
+class ActorDetail(
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.DestroyModelMixin,
+    generics.GenericAPIView,
+):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get(self, request: Request, *args, **kwargs) -> Response:
+        return self.retrieve(request, *args, **kwargs)
+
+    def put(self, request: Request, *args, **kwargs) -> Response:
+        return self.update(request, *args, **kwargs)
+
+    def patch(self, request: Request, *args, **kwargs) -> Response:
+        return self.update(request, partial=True, *args, **kwargs)
+
+    def delete(self, request: Request, *args, **kwargs) -> Response:
+        return self.destroy(request, *args, **kwargs)
```

Comment:
Mixins already take care of this for you, making the code redundant.

Reply:
No. Mixins take care of this methods only with GenericViewSet. With GenericAPIView they only let you use this simplified srtucture.
![image](https://github.com/user-attachments/assets/f77db2a0-ed97-4704-ab27-e1fd83f0ff01)

Reply:
@Alexandr-Politov actually for both generic classes you should use mixins. Mixins define actions (list, create, destroy,..) why Viewset and View have (get, post, delete,...) methods. But they aren't connected. By default DRF doesn't know that for "def post" it should call "create" action. This is why we need to do it explicitly.
Or we can inherit from ModelViewset/ListCreateAPIView/RetrieveUpdateDestroyView that already have connected methods with actions. The best idea it to view DRF code by opening these classes and check their implementation ;)